src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -487,3 +492,101 @@ def _stream_trace(streaming_url, user_name, password):
+    digest_algorithm = 'sha1'
+    thumbprint = cert.digest(digest_algorithm).decode("utf-8").replace(':', '')
+    return thumbprint
+
+def list_ssl_certs(resource_group_name):
+    client = web_client_factory()
+    return client.certificates.get_certificates(resource_group_name)
+
+def delete_ssl_cert(resource_group_name, certificate_thumbprint):
+    client = web_client_factory()
+    cert_name = None
+    webapp_certs = client.certificates.get_certificates(resource_group_name)
+    for webapp_cert in webapp_certs:
+        if webapp_cert.thumbprint == certificate_thumbprint:
+            cert_name = webapp_cert.name
+            break
+
+    return client.certificates.delete_certificate(resource_group_name, cert_name)
+
+def should_use_deployment_slot(webapp_name, slot=None):
+    qualified_site_name = None
+    if slot is not None:
+        qualified_site_name = "{}({})".format(webapp_name, slot)
+
+    return qualified_site_name
+
+def update_host_name_ssl_state(resource_group_name, webapp_name, location,
+                               host_name, ssl_state, thumbprint, client, slot=None):
+    webappWithNewSslBinding = Site(host_name_ssl_states=
+                                   [HostNameSslState
+                                    (
+                                        name=host_name,
+                                        ssl_state=ssl_state,
+                                        thumbprint=thumbprint,
+                                        to_update=True
+                                    )
+                                   ],
+                                   location=location)
+    if should_use_deployment_slot(webapp_name) is None:
+        return client.sites.create_or_update_site(resource_group_name,
+                                                  webapp_name,
+                                                  site_envelope=webappWithNewSslBinding)
+    else:
+        return client.sites.create_or_update_site_slot(resource_group_name,
+                                                       webapp_name,
+                                                       site_envelope=webappWithNewSslBinding,
+                                                       slot=slot)
+
+def update_ssl_binding(resource_group_name, name, certificate_thumbprint, ssl_type, slot=None):
+    client = web_client_factory()
+    webapp = client.sites.get_site(resource_group_name, name)
+    host_name = None
+    webapp_certs = client.certificates.get_certificates(resource_group_name)
+    for webapp_cert in webapp_certs:
+        if webapp_cert.thumbprint == certificate_thumbprint:
+            host_name = webapp_cert.host_names[0]
+    return update_host_name_ssl_state(resource_group_name, name, webapp.location,
+                                      host_name, ssl_type, certificate_thumbprint, client, slot)
+
+def bind_ssl_cert(resource_group_name, name, certificate_thumbprint, ssl_type, slot=None):
+    ssl_type_str = ssl_type.upper()
```

Comment:
If you are using **enum_choice_list for the `ssl_type` field and reflecting on your enum, you don't need to do this. You will get the value and casing as specified in the enum, regardless of what casing the user used.